common/src/main/java/com/wynntils/commands/WynntilsCommand.java

```diff
@@ -4,36 +4,319 @@
+import com.mojang.brigadier.builder.LiteralArgumentBuilder;
 import com.mojang.brigadier.context.CommandContext;
+import com.mojang.brigadier.suggestion.SuggestionProvider;
 import com.wynntils.core.Reference;
 import com.wynntils.core.commands.CommandBase;
+import com.wynntils.core.config.properties.Config;
 import com.wynntils.core.features.Feature;
 import com.wynntils.core.features.FeatureRegistry;
 import com.wynntils.core.webapi.WebManager;
 import com.wynntils.mc.utils.McUtils;
+import com.wynntils.utils.objects.CustomColor;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
 import net.minecraft.ChatFormatting;
 import net.minecraft.commands.CommandSourceStack;
 import net.minecraft.commands.Commands;
+import net.minecraft.commands.SharedSuggestionProvider;
 import net.minecraft.network.chat.ClickEvent;
 import net.minecraft.network.chat.HoverEvent;
 import net.minecraft.network.chat.MutableComponent;
 import net.minecraft.network.chat.Style;
 import net.minecraft.network.chat.TextComponent;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.reflect.FieldUtils;
 
 public class WynntilsCommand extends CommandBase {
     @Override
     public void register(CommandDispatcher<CommandSourceStack> dispatcher) {
+        LiteralArgumentBuilder<CommandSourceStack> getConfigArgBuilder = Commands.literal("get");
+        LiteralArgumentBuilder<CommandSourceStack> setConfigArgBuilder = Commands.literal("set");
+
+        SuggestionProvider<CommandSourceStack> featureSuggestionProvider =
+                (context, builder) -> SharedSuggestionProvider.suggest(
+                        FeatureRegistry.getFeatures().stream().map(Feature::getShortName), builder);
+
+        SuggestionProvider<CommandSourceStack> featureConfigSuggestionProvider =
+                (context, builder) -> SharedSuggestionProvider.suggest(
+                        () -> {
+                            String featureName = context.getArgument("feature", String.class);
+
+                            Optional<Feature> foundFeature = FeatureRegistry.getFeatureFromString(featureName);
+
+                            if (foundFeature.isEmpty()) return Collections.emptyIterator();
+
+                            return Arrays.stream(foundFeature.get().getConfigFields())
+                                    .map(Field::getName)
+                                    .iterator();
+                        },
+                        builder);
+
+        // Feature specified, config option is not, print all configs
+        // If a feature and config field is specified, print field specific info
+        // /wynntils config get <feature>
+        // /wynntils config get <feature> <field>
+        getConfigArgBuilder.then(Commands.argument("feature", StringArgumentType.word())
```

Comment:
is there a way to define all these command suggestions in their own methods or private classes? i don't really like how they're taking up space here in the registration method

Reply:
We can make the suggestion providers fields and move registration to their own methods I think.

Reply:
Yes, the WyntilsCommand class is getting quite cluttered. It would be nice if we could break it down into the "subcommand" components. I'm not against creating a `wynntils.commands.wynntils` package and put like `ConfigSubcommand.java` in there, and just let `WynntilsCommand` dispatch away. Don't know if it's feasible.

Reply:
I will not require that you do that in this PR.

Reply:
It's not a big change so I'll do this alongside the better type conversation system.